### PR TITLE
Python 3.11 and 3.12 support

### PR DESCRIPTION
**Pull request by ebattat**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description

Adding Python 3.11/3.12 support
Update PyYAML==6.0 to yYAML==6.0.1 for support Python3.12

### CODE REVIEW

**Review by athiruma — APPROVED** [review]
/lgtm

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,4 +14,4 @@ repos:
     rev: v4.4.0
     hooks:
       - id: check-json
-      - id: check-yaml
+      - id: check-yaml
```
Comment (RobertKrawitz) [inline .pre-commit-config.yaml:17]: Needs a trailing newline (i. e. undo this change).

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
Just one trivial issue.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ebattat) [reply .pre-commit-config.yaml:17]: Done

**Review by RobertKrawitz — APPROVED** [review]

**Comment by ebattat** [comment]
Here are all the PRs that are involved in the Python 3.12 upgrade:
https://github.com/redhat-performance/benchmark-runner/pull/823
https://github.com/redhat-performance/benchmark-runner/pull/831
https://github.com/redhat-performance/benchmark-runner/pull/832
https://github.com/redhat-performance/benchmark-runner/pull/845
